webpack/assets/javascripts/react_app/components/Editor/components/EditorNavbar.js

```diff
@@ -49,117 +52,143 @@ const EditorNavbar = ({
+
+  const selectedRenderPath = useMemo(
+    () => (safemode ? safemodeRenderPath : renderPath),
+    [safemode, safemodeRenderPath, renderPath]
+  );
+
+  useEffect(() => {
+    previewTemplate({ host: selectedHost, renderPath: selectedRenderPath });
+  }, [selectedRenderPath]);
+
+  return (
+    <div className="navbar navbar-form navbar-full-width navbar-editor">
+      <Nav className="nav nav-tabs nav-tabs-pf nav-tabs-pf-secondary">
+        <EditorRadioButton
+          stateView={selectedView}
+          btnView="input"
+          title={__('Editor')}
+          onClick={() => {
+            if (selectedView !== 'input') {
+              if (isRendering) toggleRenderView();
+              changeTab('input');
+            }
+          }}
+        />
+        <EditorRadioButton
+          stateView={selectedView}
+          disabled={!isDiff}
+          btnView="diff"
+          title={__('Changes')}
+          onClick={() => {
+            if (selectedView !== 'diff') {
+              changeTab('diff');
+            }
+          }}
+        />
+        {showPreview && (
+          <React.Fragment>
+            <EditorRadioButton
+              stateView={selectedView}
+              btnView="preview"
+              title={__('Preview')}
+              onClick={() => {
+                if (selectedView !== 'preview') {
+                  if (!isRendering) toggleRenderView();
+                  changeTab('preview');
+                  if (selectedHost.id === '')
+                    fetchAndPreview(selectedRenderPath);
+                }
+              }}
             />
-          )}
-          {selectedView === 'preview' &&
-            previewResult !== '' &&
-            renderedEditorValue !== value && (
-              <div id="outdated-preview-alert">
-                <Alert type="warning">
-                  {__('Preview is outdated.')}
-                  <Button
-                    bsStyle="link"
-                    onClick={() =>
-                      previewTemplate({ host: selectedHost, renderPath })
-                    }
-                  >
-                    {__('Preview')}
-                  </Button>
-                </Alert>
+            {showHostSelector && (
+              <EditorHostSelect
+                show={selectedView === 'preview'}
+                open={isSelectOpen}
+                selectedItem={selectedHost}
+                placeholder={__('Select Host...')}
+                isLoading={isFetchingHosts}
+                onChange={host =>
+                  previewTemplate({ host, renderPath: selectedRenderPath })
+                }
+                searchQuery={searchQuery}
+                onToggle={onHostSelectToggle}
+                onSearchChange={onHostSearch}
+                onSearchClear={onSearchClear}
+                options={isSearchingHosts ? filteredHosts : hosts}
+                key="hostsSelect"
+              />
+            )}
+            {!isSafemodeEnabled && (
```

Comment:
I think the proper UX would keep the option there, although disabled if safemode setting is set.

Reply:
@ShimShtein I think this feature is about migration to safemode and if you are already using safemode, you should not notice any changes. I would vote to keep the UI simple. WDYT?

Reply:
I concur. The preferred mode is safemode on, so we should not encourage people to disable it.

Reply:
I also think it's a good idea to display it as a disabled checkbox if the setting is on,
we have this behavior on many pages in Foreman, specially with buttons that are shown but disabled based on a setting or a response from server.

IMO it would help the users to be familiar with this cool new option in the Editors.

Reply:
> I concur. The preferred mode is safemode on, so we should not encourage people to disable it.

Since we disable the control, it becomes only a reminder that the setting is on (which is good, since it explains a couple of possible errors that can pop up to the user).
I think since the user will not be able to turn the setting off from the screen and he has to perform extra steps to turn  the setting off, it's enough of a barrier to prevent the user from doing it.

Reply:
the Safemode checkbox is now visible, even if Safemode rendering is enabled in the settings